p2p/conn/connection_test.go

```diff
@@ -771,16 +771,22 @@ func stopAll(t *testing.T, stoppers ...stopper) func() {
 	}
 }
 
-// generateMessages sends a sequence of messages to the specified multiplex connection `mc`.
+// generateAndSendMessages sends a sequence of messages to the specified multiplex connection `mc`.
 // Each message has the given size and is sent at the specified rate
 // `messagingRate`. This process continues for the duration `totalDuration` or
 // until `totalNum` messages are sent. If `totalNum` is negative,
 // messaging persists for the entire `totalDuration`.
-func generateMessages(mc *MConnection,
+func generateAndSendMessages(mc *MConnection,
 	messagingRate time.Duration,
-	totalDuration time.Duration, totalNum int, msgSize int, chID byte) {
+	totalDuration time.Duration, totalNum int, msgSize int,
+	msgContnet []byte, chID byte) {
+	var msg []byte
 	// all messages have an identical content
```

Comment:
[optional] proposal to delete this line because it's not always true

```suggestion
```

or move it inside the branch it applies to

```go
	if msgContnet == nil {
		// all messages have an identical content
		msg = bytes.Repeat([]byte{'x'}, msgSize)
	} else {
		msg = msgContnet
	}
```

Reply:
Done https://github.com/celestiaorg/celestia-core/pull/1179/commits/91265109e6225e34a9f2955d6f52c12c9c2b4f0e